autorest/azure/async.go

```diff
@@ -428,6 +428,7 @@ func (pt *pollingTrackerBase) pollForStatus(sender autorest.Sender) error {
 	} else {
 		// check response body for error content
 		pt.updateErrorFromResponse()
+		err = pt.pollingError()
```

Comment:
to be clear, it is still valid for the polling to fail and not return an error ? is there any error format prescribed on polling ?

Reply:
No, if the polling fails it should now always return a non-nil error.  If the service returns an error it should be in OData v4 format.

Reply:
aha ok. is this documented for polling errors somewhere ?

Reply:
Yes see https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#operation-resource-format